src/odbc/sql2tds.c

```diff
@@ -256,9 +263,11 @@ odbc_convert_table(TDS_STMT *stmt, SQLTVP *src, TDS_TVP *dest, SQLLEN num_rows)
 		if (!(new_params = tds_alloc_param_result(params)))
 			return TDS_CONVERT_NOMEM;
 
-		odbc_convert_table_row(apd, ipd, j + 1, cols[j]->fParamType, cols[j]->fCType,
+		res = odbc_convert_table_elem(apd, ipd, j + 1, cols[j]->fParamType, cols[j]->fCType,
 			cols[j]->fSqlType, cols[j]->cbColDef, cols[j]->ibScale,
 			NULL, cols[j]->cbValueMax, (SQLLEN *) SQL_NULL_DATA);
+		if (res != TDS_SUCCESS)
+			return res;
```

Comment:
Apparently you used 2 types of enumerations here. Conversion errors and TDS errors.